commands/show.go

```diff
@@ -143,6 +165,245 @@ func runShowBug(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+func showPlainFormatter(snapshot *bug.Snapshot) error {
+	// Header
+	fmt.Printf("%s [%s] %s\n",
+		snapshot.Id().Human(),
+		snapshot.Status,
+		snapshot.Title,
+	)
+
+	fmt.Printf("author: %s\n",
+		snapshot.Author.DisplayName(),
+	)
+
+	fmt.Printf("creation time: %s\n",
+		snapshot.CreatedAt.String(),
+	)
+
+	fmt.Printf("last edit: %s\n",
+		snapshot.LastEditTime().String(),
+	)
+
+	// Labels
+	var labels = make([]string, len(snapshot.Labels))
+	for i := range snapshot.Labels {
+		labels[i] = string(snapshot.Labels[i])
+	}
+
+	fmt.Printf("labels: %s\n",
+		strings.Join(labels, ", "),
+	)
+
+	// Actors
+	var actors = make([]string, len(snapshot.Actors))
+	for i := range snapshot.Actors {
+		actors[i] = snapshot.Actors[i].DisplayName()
+	}
+
+	fmt.Printf("actors: %s\n",
+		strings.Join(actors, ", "),
+	)
+
+	// Participants
+	var participants = make([]string, len(snapshot.Participants))
+	for i := range snapshot.Participants {
+		participants[i] = snapshot.Participants[i].DisplayName()
+	}
+
+	fmt.Printf("participants: %s\n",
+		strings.Join(participants, ", "),
+	)
+
+	// Comments
+	indent := "  "
+
+	for i, comment := range snapshot.Comments {
+		var message string
+		fmt.Printf("%s#%d %s <%s>\n",
+			indent,
+			i,
+			comment.Author.DisplayName(),
+			comment.Author.Email(),
+		)
+
+		if comment.Message == "" {
+			message = "No description provided."
+		} else {
+			message = comment.Message
+		}
+
+		fmt.Printf("%s%s\n",
+			indent,
+			message,
+		)
+	}
+
+	return nil
+}
+
+type JSONBugSnapshot struct {
+	Id           string    `json:"id"`
+	HumanId      string    `json:"human_id"`
+	CreationTime time.Time `json:"creation_time"`
+	LastEdited   time.Time `json:"last_edited"`
+
+	Status       string         `json:"status"`
+	Labels       []bug.Label    `json:"labels"`
+	Title        string         `json:"title"`
+	Author       JSONIdentity   `json:"author"`
+	Actors       []JSONIdentity `json:"actors"`
+	Participants []JSONIdentity `json:"participants"`
+
+	Comments []JSONComment `json:"comments"`
+}
+
+type JSONComment struct {
+	Id          int    `json:"id"`
+	AuthorName  string `json:"author_name"`
+	AuthorLogin string `json:"author_login"`
+	Message     string `json:"message"`
+}
+
+func showJsonFormatter(snapshot *bug.Snapshot) error {
+	jsonBug := JSONBugSnapshot{
+		snapshot.Id().String(),
+		snapshot.Id().Human(),
+		snapshot.CreatedAt,
+		snapshot.LastEditTime(),
+		snapshot.Status.String(),
+		snapshot.Labels,
+		snapshot.Title,
+		JSONIdentity{},
+		[]JSONIdentity{},
+		[]JSONIdentity{},
+		[]JSONComment{},
+	}
+
+	jsonBug.Author.Name = snapshot.Author.DisplayName()
```

Comment:
instead of repeating that, you can make yourself a small helper:
```go
makeIdentity := func(i *identity.Interface) JSONIdentity {
    return JSONIdentity{
      ...
    }
}

...

jsonBug.Author = makeIdentity(snapshot.Author)
```

Reply:
As this is also used elsewhere you can make that a real function `newJSONIdentity` next to JSONIdentity.